rotkehlchen/balances/historical.py

```diff
@@ -120,6 +121,67 @@ def get_assets_amounts(
 
         return amounts, last_event_id
 
+    def get_historical_netvalue(
+            self,
+            from_ts: Timestamp,
+            to_ts: Timestamp,
+    ) -> tuple[dict[Timestamp, FVal], list[tuple[str, Timestamp]]]:
+        """Calculates historical net worth per day within the given time range.
+
+        Uses asset balances per day combined with historical prices to calculate
+        the total net worth in the user's main currency for each day.
+
+        May raise:
+            - NotFoundError if no events exist in the specified time period.
+
+        Returns:
+            - A mapping of timestamps to net worth in main currency for each day
+            - A list of (asset id, timestamp) tuples where price data was missing
+        """
+        events, main_currency = self._get_events_and_currency(from_ts=from_ts, to_ts=to_ts)
```

Comment:
Regarding this function it is very expensive since it loads all the events into memery and initializes a lot of stuff that we ignore. It would be good to have a lighter version but I would add it as a todo here

Reply:
added a comment here